### PR TITLE
Implement data explorer tool

### LINKED ISSUE #2153: Implement data explorer tool
### Description

Allow students to choose which columns of a table to be sent to a graph. Also allow them to choose the type of graph.

### PR DESCRIPTION
**Pull request by geoffreykwan**
Test the data explorer feature in the Table component. Run ```npm install``` to download new client libraries.

1. Log in as a teacher
1. Open a project in the Authoring Tool
1. Create a step
1. Add a Table component
1. Add a Graph component
1. In the Graph component clear out the Units, Min, and Max fields for the X Axis and Y Axis. Also set the "Round Values To" parameter to "No Rounding".
1. Add a Connected Component and make it point to the Table component in the same step. Set the Type to "Show Work". Leave the other fields empty.
1. In the Table component go to the Advanced section and enable the Data Explorer. Change the Data Explorer options and preview the project.
1. Make sure all the graph types work
1. Make sure the regression line works for the scatter plot
1. Make sure multiple number of series works
1. Make sure the student can edit the axis labels if the author allows them to

Closes #2153

### CODE REVIEW

File: src/main/webapp/wise5/components/graph/graphController.js [hunk 1]
```diff
@@ -253,9 +262,156 @@ class GraphController extends ComponentController {
...
+      const xColumn = dataExplorerSeries[seriesIndex].xColumn;
+      const yColumn = dataExplorerSeries[seriesIndex].yColumn;
+      if (yColumn != null) {
+        const color = this.dataExplorerColors[seriesIndex];
+        const name = dataExplorerSeries[seriesIndex].name;
+        const series = this.generateDataExplorerSeries(studentData, xColumn, yColumn, graphType,
+            name, color);
+        trial.series.push(series);
+        if (graphType === 'scatter' && studentData.isDataExplorerScatterPlotRegressionLineEnabled) {
+          const regressionSeries = this.generateDataExplorerRegressionSeries(studentData, xColumn,
```
[the remark this answers is not in the document]
Comment (breity) [reply src/main/webapp/wise5/components/graph/graphController.js]: @hirokiterashima What's the rationale for only allowing integers in cell values? Wouldn't this be limiting?

**Comment by breity** [comment]
@hirokiterashima What's the rationale for only allowing integers in cell values? Wouldn't this be limiting?

**Comment by hirokiterashima** [comment]
@breity the restriction was mostly to guard against user errors, like when a student types words in a number-only cell. Highcharts seems to ignore the row data in this case, so we probably don't need the check.

**Comment by hirokiterashima** [comment]
latest changes look good!

**Comment by breity** [comment]
@hirokiterashima Ah, you mean numbers vs strings. I was talking about integers vs numbers. (Because lots of data sets will use non-integer numbers.)

Good that Highcharts handles mismatched data types anyways. In the future we will also want to support categorical variables that could be strings.

**Pull request merged**